src/components/NewMessageForm/AudioRecorder/AudioRecorder.vue

```diff
@@ -0,0 +1,340 @@
+			const minutes = this.recordTime.minutes.toString().length === 2 ? this.recordTime.minutes : `0${this.recordTime.minutes}`
+			return `${minutes}:${seconds}`
+		},
+
+		tooltipDelay() {
+			return { show: 500, hide: 200 }
+		},
+
+		startRecordingTooltip() {
+			return t('spreed', 'Record voice message')
+		},
+
+		stopRecordingTooltip() {
+			return t('spreed', 'End recording and send')
+		},
+
+		abortRecordingTooltip() {
+			return t('spreed', 'Dismiss recording')
+		},
+	},
+
+	watch: {
+
+		isRecording(newValue) {
+			console.debug('isRecording', newValue)
+		},
+	},
+
+	methods: {
+		/**
+		 * Initialize the media stream and start capturing the audio
+		 */
+		async start() {
+			// Create new audio stream
+			try {
+				this.audioStream = await mediaDevicesManager.getUserMedia({
+					audio: true,
+				})
+			} catch (exception) {
+				console.debug(exception)
+				if (exception.name === 'NotAllowedError') {
+					showError(t('spreed', 'Access to the microphone was denied'))
+				} else {
+					showError(t('spreed', 'Microphone either not available or disabled in settings'))
+				}
+				return
+			}
+
+			// Create a mediarecorder to capture the stream
+			try {
+				this.mediaRecorder = new MediaRecorder(this.audioStream)
+			} catch (exception) {
+				console.debug(exception)
+				this.audioStream.getTracks().forEach(track => track.stop())
+				this.audioStream = null
+				showError(t('spreed', 'Error while recording audio'))
+				return
+			}
+
+			// Add event handler to onstop
+			this.mediaRecorder.onstop = this.generateFile
+
+			// Add event handler to ondataAvailable
+			this.mediaRecorder.ondataavailable = (e) => {
+				this.chunks.push(e.data)
+			}
+
+			try {
+				// Start the recording
+				this.mediaRecorder.start()
+			} catch (exception) {
+				console.debug(exception)
+				this.aborted = true
+				this.stop()
+				this.resetComponentData()
+				showError(t('spreed', 'Error while recording audio'))
+				return
+			}
+
+			console.debug(this.mediaRecorder.state)
+
+			// Start the timer
+			this.recordTimer = setInterval(() => {
+				if (this.recordTime.seconds === 59) {
+					this.recordTime.minutes++
+					this.recordTime.seconds = 0
+				}
+				this.recordTime.seconds++
+			}, 1000)
+			// Forward an event to let the parent NewMessageForm component
+			// that there's an undergoing recording operation
+			this.$emit('recording', true)
+		},
+
+		/**
+		 * Stop the mediaRecorder
+		 */
+		stop() {
+			this.mediaRecorder.stop()
+			clearInterval(this.recordTimer)
+			this.$emit('recording', false)
+		},
+
+		/**
+		 * Generate the file
+		 */
+		generateFile() {
+			this.audioStream.getTracks().forEach(track => track.stop())
+			if (!this.aborted) {
+				this.blob = new Blob(this.chunks, { 'type': 'audio/mpeg-3' })
+				// Convert blob to file
+				const audioFile = new File([this.blob], `${Date.now()}.mp3`)
```

Comment:
I would vote for a file name like:
`REC-YYYY-MM-DD_HH-MM-SS.mp3`
Maybe we even add the conversation name? Could help with search later on?
`REC-YYYY-MM-DD_HH-MM-SS_{{ conversationName }}.mp3`

Reply:
This will show up in Files, right? So it would be great to go for a slightly less technical name (unlike these IMG_1234 like names or "REC".

To compare with the GNOME screenshot app, we could use: "Recording from 2021-06-08 11-05-01.mp3". And yeah we could include the conversation name:
> Recording from 2021-06-08 11-05-01 (conversation name).mp3

Reply:
![time](https://user-images.githubusercontent.com/26852655/121364524-c3c44d80-c92f-11eb-9232-fe3d07ae907e.PNG)

Reply:
Sounds good?

Reply:
See https://github.com/nextcloud/spreed/pull/5610#discussion_r648320400